packages/web-components/@sfx/search-box/src/search-box.ts

```diff
@@ -0,0 +1,194 @@
+   */
+  @property({ type: String }) value: string = '';
+  /**
+   * Determines whether or not the search button is present.
+   */
+  @property({ type: Boolean, reflect: true })
+  searchButton: boolean = false;
+  /**
+   * Determines whether or not the clear button is present.
+   */
+  @property({ type: Boolean, reflect: true })
+  clearButton: boolean = false;
+
+  constructor() {
+    super();
+    this.updateText = this.updateText.bind(this);
+  }
+
+  /**
+   * Adds event listeners.
+   */
+  connectedCallback() {
+    super.connectedCallback();
+    window.addEventListener(SEARCHBOX_EVENT.UPDATE_SEARCH_TERM, this.updateText);
+  }
+
+  /**
+   * Removes event listeners.
+   */
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    window.removeEventListener(SEARCHBOX_EVENT.UPDATE_SEARCH_TERM, this.updateText);
+  }
+
+  /**
+   * Dispatches a search request event with the `value` property.
+   * Invoked in response to user interactions: `enter` key or click on search button.
+   */
+  emitSearchEvent() {
+    const searchboxRequestEvent = new CustomEvent(SEARCHBOX_EVENT.SEARCH_REQUEST, {
+      detail: this.value,
+      bubbles: true
+    });
+    this.dispatchEvent(searchboxRequestEvent);
+  }
+
+  /**
+   * Dispatches an event notifying that the input box has been cleared.
+   * Invoked in response to a click on the clear button.
+   */
+  emitSearchBoxClearClick() {
+    const searchboxClearedEvent = new CustomEvent(SEARCHBOX_EVENT.SEARCHBOX_CLEAR_CLICK, { bubbles: true });
+    this.dispatchEvent(searchboxClearedEvent);
+  }
+
+  /**
+   * Updates the contents of the search input box and the value property
+   * with the payload of the given event.
+   * Invoked in response to an update search term event.
+   *
+   * @param e The event object.
+   */
+  updateText(e: CustomEvent) {
+    this.getInputElement().value = e.detail;
+    this.updateSearchTermValue(e.detail);
+  }
+
+  /**
+   * Returns the searchbox input element
+   */
+  getInputElement(): HTMLInputElement {
+    return this.querySelector('[data-sfx-ref=searchInput]') as HTMLInputElement;
+  }
+
+  /**
+   * Dispatches an `sfx::on_search_box_change` event, and updates the value
+   * property with the payload of the given event.
+   *
+   * Invoked in response to a change to the contents of the searchbox.
+   *
+   * @param e The KeyboardEvent object.
+   */
+  handleInput(e: KeyboardEvent) {
+    this.updateSearchTermValue((e.target as HTMLInputElement).value);
+    this.dispatchEvent(
+      new CustomEvent(SEARCHBOX_EVENT.SEARCHBOX_CHANGE, {
+        detail: (e.target as HTMLInputElement).value,
+        bubbles: true
+      })
+    );
+  }
+
+  /**
+   * Invokes the emitSearchEvent function when a user presses enter and 
+   * the value property length is greater than 0.
+   *
+   * @param e The KeyboardEvent object.
+   */
+  handleKeydown(e: KeyboardEvent) {
+    if (e.key === "Enter" && this.value.length > 0) {
```

Comment:
I like double quotes, but we should have single quotes for consistency.

```suggestion
    if (e.key === 'Enter' && this.value.length > 0) {
```